src/braket/circuits/circuit.py

```
@@ -643,6 +657,69 @@ def add_verbatim_box(
+
+        Examples:
+            >>> circ = Circuit.h(0).cnot(0, 1).measure([0])
+            >>> circ.print(list(circ.instructions))
+            [Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(0)]),
+            Instruction('operator': CNot('qubit_count': 2), 'target': QubitSet([Qubit(0),
+                Qubit(1)]),
+            Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(2)]),
+            Instruction('operator': Measure, 'target': QubitSet([Qubit(0)])]
+        """
+        # check whether measuring an empty circuit
+        if not self.qubits:
+            raise IndexError("Cannot measure an empty circuit.")
+
+        if type(target_qubits) is int:
+            target_qubits = [target_qubits]
+
+        # Check that the target qubits are on the circuit
+        if target_qubits and not all(qubit in self.qubits for qubit in target_qubits):
+            raise IndexError("Target qubits must be within the range of the current circuit.")
+
+        # Check if result types are added on the circuit
+        if self.result_types:
+            raise ValueError("a circuit cannot contain both measure instructions and result types.")
+
+        if target_qubits:
+            self.add_instruction(
+                Instruction(
+                    operator=Measure(
+                        qubit_count=len(target_qubits), ascii_symbols=["M"] * len(target_qubits)
+                    ),
+                    target=target_qubits,
+                )
+            )
+            self._measure_targets = target_qubits
+        else:
+            # Measure all the qubits
+            self.add_instruction(
+                Instruction(
+                    operator=Measure(
+                        qubit_count=self.qubit_count, ascii_symbols=["M"] * self.qubit_count
+                    ),
+                    target=self.qubits,
+                )
+            )
+            self._measure_targets = self.qubits
```

Similarly, I think we should throw an error here if `self._measure_targets` already contains any of the qubits. Given that we don't allow gates after measurement, it doesn't make sense to allow multiple measurements of the same qubit.

---

Added an error here if there are any `self._measure_targets` specified to prevent cases like `Circuit.h(0).measure(0).measure()`